### DISCUSSION TITLE
I'm trying to clean this up

### DISCUSSION THREAD

**Discussion by endolith** [endolith/waveform-analysis#31] [Announcements]
I made it a long time ago and much of it doesn't work and it's poorly organized, but now it's somehow my most popular repo. Sorry.

**Comment by MirkoThulke** [comment]
Thank you for the implementation of this filter . I had a lesson learned on my side : When filtering audio chunk-wise, it is important to carryover the filter state from chunk filtering to chunk+1 . Otherwise the filter output is suffering from noise . Maybe you could add a hint on your wikipage . I also wrote a small unit test to check if the filter works fine . See my repo .
